compiler/packages/babel-plugin-react-compiler/src/HIR/CollectHoistablePropertyLoads.ts

```diff
@@ -0,0 +1,520 @@
+import {CompilerError} from '../CompilerError';
+import {inRange} from '../ReactiveScopes/InferReactiveScopeVariables';
+import {Set_intersect, Set_union, getOrInsertDefault} from '../Utils/utils';
+import {
+  BasicBlock,
+  BlockId,
+  DeclarationId,
+  GeneratedSource,
+  HIRFunction,
+  Identifier,
+  Place,
+  PropertyLoad,
+  ReactiveScopeDependency,
+  ScopeId,
+  TInstruction,
+} from './HIR';
+
+export type CollectHoistablePropertyLoadsResult = {
+  nodes: ReadonlyMap<ScopeId, BlockInfo>;
+  temporaries: ReadonlyMap<Identifier, Identifier>;
+  properties: ReadonlyMap<Identifier, ReactiveScopeDependency>;
```

Comment:
For these sidemaps, any reason to key it by the identifier itself and not an identifierID?

Reply:
Updated to use `IdentifierId` and double checked that this changed nothing internally on 100k+ files